.github/workflows/rust.yml

```diff
@@ -11,13 +11,23 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout repo
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
     - name: Install Rust
-      uses: dtolnay/rust-toolchain@1.80.0
+      uses: dtolnay/rust-toolchain@stable
       with:
         components: clippy
     - name: Run cargo-deny
       uses: EmbarkStudios/cargo-deny-action@v1
     - run: cargo clippy -- -W clippy::pedantic
-    - run: cargo test
+    - run: cargo fmt --all --check
+    - run: cargo test --all-targets
     - run: WASTEBIN_BASE_URL="http://127.0.0.1:8080/wastebin" cargo test  # port is not relevant
+
+  spaces-check:
+    name: Spaces Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Spaces check
+        run: git diff-tree --check $(git hash-object -t tree /dev/null) HEAD
```

Comment:
Could you please explain what this does?

Reply:
Checks for introduction of white space errors, see https://git-scm.com/docs/git-diff-tree#Documentation/git-diff-tree.txt---check.